### DISCUSSION TITLE
Load data from local file

### DISCUSSION THREAD

**Discussion by Davdup** [quasarframework/quasar#12990] [CLI - Electron mode]
Hi,
I am building a small application for a client (electron mode), and I would like him to be able to load some piece of information from a local/editable file (stored in "resources"?): js file, json, csv, excel (the most common the better, ie excel preferrably!). Then the application would read this data from this file (either automatically upon start, or at user's request) and do its job to make calculations based on this information...
Any .js file I am importing in my vue pages is "packaged" at the app building stage and is no more accessible to the user.
Can someone point me to the right direction??
Thanks!

David

**Comment by mJeromeDiaz** [comment]
Hi @Davdup, 
i dont understand your problems.
you can add an file picker (https://quasar.dev/vue-components/file-picker) and edit file content in your vue file no ?

**Reply by Davdup** [reply]
Let me be more specific: my Electron application will calculate costs of projects based on a list of item prices that is confidential, and that I cannot hard code into the project (like any other .js file that I can use inside my Application). So I want the user to be able for example to replace an accessible (=public) file that would contain sample data (fake prices) with his own confidential file (containing the real prices, but that he only could view on his computer). So I would like to be able to load an external file as an array data... I hope this clarifies my request?

**Reply by Davdup** [reply]
Or perhaps I can use a json file stored somewhere it will not be processed during the app building process and just copied to an accessible location (like assets? public??). Would that be possible?

**Reply by mJeromeDiaz** [reply]
I understand, you can just add an filepicker, edit file content et delete file when work is over ;)

**Reply by Davdup** [reply]
Hi, thanks for your effort, but that's not straight forward at all for me, I'm not an expert at all in coding, and this is apparently out of my reach... I was trying thigs with no success at all...
If I write it in understandable language, I would like to (from a .vue page in my Quasar App, in Electron mode)

- when clicking somewhere, read that content of a JSON file stored in /public forlder, file that remains accessible to the user once the Electron App is built (but not packaged as an asar file)
- Use the content of this file as a javascript object to perform some calculations on my .vue page

Can anyone point me to an example that does that or something similar??

Thanks!

David
